### PR TITLE
Implemented GetLeavingArcsIndexBatches and GetEnteringArcsIndexBatch

### PR DESCRIPTION
**Pull request by qindazhu**
- Implemented `GetLeavingArcsIndexBatches` and `GetEnteringArcIndexBatches` and tested
- Tested `GetIncomingArcs`

Will add random size tests after implement other fsa_utils algorithms (but may not so hurry as these algorithms would have no very complex kernel calls)

Noted those files except `fsa_utils*.*` are just style fixes.

### CODE REVIEW

File: k2/csrc/fsa_utils.cu [hunk 1]
```diff
@@ -814,4 +814,103 @@ Ragged<int32_t> GetIncomingArcs(FsaVec &fsas,
       incoming_arcs_order);
 }
 
+Ragged<int32_t> GetLeavingArcIndexBatches(FsaVec &fsas,
+                                          Ragged<int32_t> &state_batches) {
+  K2_CHECK_EQ(fsas.NumAxes(), 3);
```
Comment (csukuangfj) [inline k2/csrc/fsa_utils.cu:819]: Do we need to check the number of axes with `state_batches`?

**Comment by csukuangfj** [comment]
+1

File: k2/csrc/fsa_utils.cu (hunk 1 above)
Comment (qindazhu) [reply k2/csrc/fsa_utils.cu:819]: Added

**Comment by danpovey** [comment]
Great!
Guys its been a long day and I had lots of zoom calls this evening,
Please merge  those PRs if you guys have looked and it seems appopriate to
merge, I'll try to glance at them but that's all.
Dan

**Comment by csukuangfj** [comment]
merging.

**Pull request merged**